src/node/miner.h

```diff
@@ -232,6 +234,17 @@ void ApplyArgsManOptions(const ArgsManager& gArgs, BlockAssembler::Options& opti
 
 /* Compute the block's merkle root, insert or replace the coinbase transaction and the merkle root into the block */
 void AddMerkleRootAndCoinbase(CBlock& block, CTransactionRef coinbase, uint32_t version, uint32_t timestamp, uint32_t nonce);
+
+/**
+ * Return a new block template when fees rise to a certain threshold or after a
+ * new tip; return nullopt if timeout is reached.
+ */
+std::unique_ptr<CBlockTemplate> WaitAndCreateNewBlock(ChainstateManager& chainman,
+                                                      KernelNotifications& kernel_notifications,
+                                                      CTxMemPool* mempool,
+                                                      const std::unique_ptr<CBlockTemplate>& block_template,
```

Comment:
In commit "interfaces: refactor: move `waitNext` implementation to miner" (720f201e652885b9e0aec8e62a1bf9590052b320)

Would be good to change `const std::unique_ptr<CBlockTemplate>& block_template` to just `const CBlockTemplate& block_template` since it doesn't look like it's allowed to be null.

Reply:
yeah it is not, I will do if I have to retouch.